lib/charms/mongodb/v0/mongodb.py

```diff
@@ -290,6 +294,51 @@ def step_down_primary(self) -> None:
         """Steps down the current primary, forcing a re-election."""
         self.client.admin.command("replSetStepDown", {"stepDownSecs": "60"})
 
+    def move_primary(self, new_primary: str) -> None:
+        """Forcibly moves the primary to the new primary provided.
+
+        Args:
+            new_primary: str of the ip address which the new primary should belong.
+        """
+        # Do not move a priary unless the cluster is in sync
+        rs_status = self.client.admin.command("replSetGetStatus")
+        if self.is_any_sync(rs_status):
+            # it can take a while, we should defer
+            raise NotReadyError
+
+        is_move_successful = True
+        self.set_replicaset_election_priority(priority=0.5, ignore_member=new_primary)
+        try:
+            for attempt in Retrying(stop=stop_after_delay(180), wait=wait_fixed(3)):
+                with attempt:
+                    self.step_down_primary()
+                    if self.primary() != new_primary:
+                        raise FailedToMovePrimaryError
+        except RetryError:
+            # catch all possible exceptions when failing to step down primary. We do this in order
+            # to ensure that we reset the replica set election priority.
+            is_move_successful = False
+
+        # reset all replicas to the same priority
+        self.set_replicaset_election_priority(priority=1)
```

Comment:
Sorry I don't follow 😅 

Would you mind explaining Why " if so, might be worth setting new_primary priority to 1 while setting priority to 0.5 for other members"? 

Maybe this information helps?
1. if the function fails and re-executes [the calling hook it won't trigger this call unless it is still needed to move the primary](https://github.com/canonical/mongodb-operator/blob/137878cfb7c93142f6cfc415714489a061e19ff8/src/upgrades/mongodb_upgrade.py#L240) 
2. It is dangerous to keep the priorities at 0.5, 0.5, 1 - since you're effectively "forcing" one unit to be primary and goes against HA-principles